git/intermediate_git/working_with_remotes.md

```diff
@@ -0,0 +1,93 @@
+
+~~~bash
+$ git push origin main
+$ git rebase -i --root
+$ git push --force
+$ git log
+~~~
+
+Huh, that's interesting. We don't see our fourth file on our local system. Let's check our GitHub repository to see if it's there.
+
+Oh no, we just destroyed it!  In this scenario, the danger - you could potentially destroy the work of those you're collaborating with! `git push --force` is a **very dangerous command, and it should be used with caution when collaborating with others**. Instead, you can fix your outdated history error by updating your local history using `fetch`, `merge`, and then attempting to `push` again.
+
+Let's consider a different scenario:
+
+~~~bash
+$ touch test4.md
+$ git add test4.md && git commit -m "Create fifth file"
+$ git push origin main
+$ git log
+~~~
+We look at our commit message and realize *oops*, we made a mistake. We want to undo this commit and are once again tempted to just force the push. But wait, remember, this is a **very dangerous command**. If we're ever considering using it, always check if it's appropriate and if we can use a safer command instead. If we're collaborating with others and want to *undo* a commit we just made, we can instead use `git revert`!
+
+~~~bash
+git revert HEAD
+git push origin main
+~~~
+
+Remember when we were working with HEAD, aka the current commit we're viewing, while rebasing? What this would do is it would revert the changes to HEAD! Then we would push our new commit to whichever branch we're working on, which in this example is main even though normally our work would most likely be on a feature-branch.
+
+So now that we've learned about the various dangers of `git push --force`, you're probably wondering why it exists and when to use it. A very common scenario in which developers use `git push --force` is updating pull requests. Collaborative work is covered more in depth in a separate lesson, but the take-away from this section should be that the `--force` option should be used only when you are certain that it is appropriate. There are also less common scenarios, such as when sensitive information is accidentally uploaded to a repository and you want to remove all occurrences of it.
+
+<span id='force-with-lease'>It is worth giving special mention to `git push --force-with-lease`</span>, a command which in some companies is the default option. The reason for this is that it's a fail-safe! It checks if the branch you're attempting to push to has been updated and sends you an error if it has. This gives you an opportunity to, as mentioned before, `fetch` the work and update your local repository.
+### Dangers and best practices
```

Comment:
You'll need a newline here for[ formatting reasons](https://www.theodinproject.com/lessons/preview?uuid=c0126432-aac9-412d-8c1e-79d48d6ad3e0) as well.